src/coreclr/nativeaot/Runtime/thread.cpp

```diff
@@ -817,6 +839,9 @@ void Thread::HijackReturnAddressWorker(StackFrameIterator* frameIterator, Hijack
 #endif
 
         *ppvRetAddrLocation = (void*)pfnHijackFunction;
+#if defined(TARGET_ARM64) && defined(__GNUC__)
+        *ppvRetAddrLocation = PacSignPtr(*ppvRetAddrLocation);
```

Comment:
so what happens when we do not sign the addresses? Do they go back on stack where the attacker can hijack?

Reply:
Also, why is it just for `__GNUC__`? should also do it for windows, right? Likewise in `threadsuspend.cpp`